tap_dashboard_reports/period.py

```diff
@@ -50,6 +51,28 @@ def _calculate_start_date(self):
         return self._end_date - relativedelta(days=self._periods)
 
 
+class PeriodWeek(PeriodDefault):
+    def iterate(self):
+        cursor = self._start_date
+        while cursor < self._end_date:
+            interval_start = cursor
+            interval_start = (
+                self._start_date
+                if interval_start < self._start_date
+                else interval_start
+            )
+            interval_end = previous_day_of_week(cursor)
```

Comment:
This name doesn't make sense, in should not be the previous day, it should be the last date of the week

Reply:
changed